batch/batch/server/server.py

```diff
@@ -714,18 +677,16 @@ def from_record(record, deleted=False):
                 state = 'failure'
             elif record['n_cancelled'] > 0:
                 state = 'cancelled'
-            elif not record['is_open'] and (record['n_succeeded'] == record['n_jobs']):
+            elif record['n_succeeded'] == record['n_jobs']:
```

Comment:
Above this, I think MAX_TTL can go away.

Similarly, on 896 you have `await batch.close()`, but no Batch close method